clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -429,21 +492,40 @@ public void onResponse(final ClientResponse response) {
                     long offset = offsetAndMetadata.offset();
                     Errors error = Errors.forCode(partition.errorCode());
                     if (error == Errors.NONE) {
-                        log.debug("OffsetCommit {} for partition {}", offset, tp);
+                        log.debug("OffsetCommit completed successfully for offset {} partition {}", offset, tp);
                         continue;
                     }
 
-                    if (error == Errors.TOPIC_AUTHORIZATION_FAILED) {
+                    if (error == Errors.COORDINATOR_NOT_AVAILABLE ||
```

Comment:
Is there a reason why you completely changed the structure here? The old code looked pretty good.

Reply:
I changed it to simplify it, because the previous `CommitReqMgr` had specific logic(ifs) for multiple errors, that truly did not need any specific action (other than failing the future and logging). 
Now we do have specific actions for the errors that need it (like the coordinator ones), but the ones that only need to log and fail go to a [single place](https://github.com/apache/kafka/blob/b7eb1081cd2d7546c47ed549ea4881b51ef96fec/clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java#L531) where we use the `error.exception` and `error.message`.